portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -118,15 +127,31 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
   }
 
+  /**
+   * Returns the email address of the user, if the user is logged in.
+   * If no user is logged in, return an empty string (however, a user 
+   * only has post access when logged in).
+   * 
+   * TODO: A user must be signed in to post a comment. However, what if there 
+   * is a bug? How should this be handled?
+   */
+  public String getUserEmail() {
+    if (userService.isUserLoggedIn()) {
+      return userService.getCurrentUser().getEmail();
+    }
+    return "";
```

Comment:
I would argue that 'null' would be a better return type here. Null is a clean and understandable way of saying "what you're looking for doesn't exist", "" is an object that you're assigning special meaning to (like a return type of '-1'), which is also generally discouraged when possible.

Reply:
Nevermind, I added error handling for this case. Thanks!